### PR TITLE
[ESIMD][NFC][DOC] Add 'restriction' section to atomic_update() doc

### PR DESCRIPTION
**Pull request by v-klochkov**

### CODE REVIEW

File: sycl/doc/extensions/supported/sycl_ext_intel_esimd/sycl_ext_intel_esimd_functions.md [hunk 1]
```diff
@@ -439,7 +439,7 @@ template <typename T, int N, int VS = 1, typename OffsetSimdViewT, typename Prop
 `(slm-sc-*)`: Stores ("scatters") the vector `vals` to shared local memory locations addressed by `byte_offsets`.  
 The parameter `byte_offset` is a vector of any integral type elements for `(usm-sc-*)`, 32-bit integer elements for `(lacc-sc-*)` and `(slm-sc-*)`, any integral type integer elements for `(acc-sc-*)` in [stateless](#statelessstateful-memory-mode) mode(default),
 and up-to-32-bit integer elements for `(acc-sc-*)` in [stateful](#statelessstateful-memory-mode) mode.  
-The optional parameter `pred` provides a `simd_mask`. If some element in `pred` is zero, then the store to the corresponding memory location is skipped.  
+The optional parameter `mask` provides a `simd_mask`. If some element in `mask` is zero, then the store to the corresponding memory location is skipped.
```
Comment (sarnex) [inline sycl/doc/extensions/supported/sycl_ext_intel_esimd/sycl_ext_intel_esimd_functions.md:442]: does the actual code in the header use `mask`? If not we should have a seperate task to make the code and the code here match

**Review by sarnex — APPROVED** [review]

File: sycl/doc/extensions/supported/sycl_ext_intel_esimd/sycl_ext_intel_esimd_functions.md (hunk 1 above)
Comment (v-klochkov) [reply sycl/doc/extensions/supported/sycl_ext_intel_esimd/sycl_ext_intel_esimd_functions.md:442]: Different functions in the code have different name `pred` or `mask`. Just needed to unify them here in this doc (at least inside every of the groups (atomic_update, gather, scatter)).

Thank you for the review.

Comment (sarnex) [reply sycl/doc/extensions/supported/sycl_ext_intel_esimd/sycl_ext_intel_esimd_functions.md:442]: imo, as a separate task we can make an internal tracker for, we should unify the code to (and then update the doc)

Comment (v-klochkov) [reply sycl/doc/extensions/supported/sycl_ext_intel_esimd/sycl_ext_intel_esimd_functions.md:442]: Well, I don't mind if such re-naming is something super-quick. 
If done, it need to be very accurate to not cause any errors due to massive renames where some place might be forgotten or due to accidental renaming in a pace where it is not supposed to be done.

**Pull request merged**